projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/upload/FileFormatDetector.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2021 VMware, Inc.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.vmware.taurus.service.upload;
+
+import org.apache.tika.config.TikaConfig;
+import org.apache.tika.detect.Detector;
+import org.apache.tika.io.TikaInputStream;
+import org.apache.tika.metadata.Metadata;
+import org.apache.tika.metadata.TikaCoreProperties;
+import org.apache.tika.mime.MediaType;
+
+import java.io.IOException;
+import java.nio.file.Path;
+
+class FileFormatDetector {
+
+  /**
+   * Match detectedType with target type: If targetType is base type (e.g text and not text/plain).
+   * It will match with base type of the detected type. If targetType is full type and subtype
+   * (text/plain) , it will match exactly.
+   *
+   * <p>Examples match("text/binary", "text/binary") returns True match("text/plain", "text")
+   * returns True match("text/sql", "text/plain") returns False match("text/plain", "application")
+   * returns False
+   *
+   * @param detectedType the type that was detected and need to be verified (must in format
+   *     "type/subtype")
+   * @param targetType the target type against which it is checked (can be either "type" or
+   *     "type/subtype")
+   * @return true or false
+   */
+  public boolean matchTypes(String detectedType, String targetType) {
+    var detectedMediaType = MediaType.parse(detectedType);
+    if (detectedMediaType == null) {
+      throw new IllegalArgumentException(
+          "detectedType must in format 'type/subtyp' but it was: " + detectedType);
+    }
+    if (targetType.contains("/")) { // compare by both type and subtype (text/plain)
+      var targetMediaType = MediaType.parse(targetType);
+      return detectedMediaType.equals(targetMediaType);
+    } else { // we compare only by type (text)
+      return detectedMediaType.getType().equals(targetType);
+    }
+  }
+
+  /**
+   * Return the detected file type Full list of file types are documented in
+   * https://tika.apache.org/1.10/formats.html#Full_list_of_Supported_Formats
+   */
+  public String detectFileType(Path filePath) throws IOException {
```

Comment:
I want to keep it as an instance instead of static so that it's more flexible to changes in the future. I did not add an interface but we can have another type of file type detector in the future. It's not really just an util class.